packages/react-native/src/components/QuilttConnector.tsx

```diff
@@ -261,8 +263,25 @@ const QuilttConnector = ({
 
             // Check if oauthUrl exists before proceeding
             if (oauthUrl) {
-              // Create a new URL from the normalized oauthUrl
-              handleOAuthUrl(oauthUrl)
+              // Check if the URL is already encoded before decoding
+              // This is because different providers might pass the OAuth URL differently
+              if (isEncoded(oauthUrl)) {
+                try {
+                  // Decode it once to prevent double decoding
+                  const decodedUrl = decodeURIComponent(oauthUrl)
+                  console.log('Decoded encoded oauthUrl:', decodedUrl)
```

Comment:
Do you need these for debugging still? The oAuthUrls contain somewhat sensitive token, so we should avoid leaking these out into the broader context of logs. 
```suggestion
```